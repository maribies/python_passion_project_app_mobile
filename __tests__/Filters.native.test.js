import React from "react";
import { ThemeProvider } from "emotion-theming";
import { create } from "react-test-renderer";
import {
  Filters,
  createFilter,
  createFilters,
} from "../Components/Filters.native";

import filters from "../Fixtures/filters.json";

const theme = {
  margin: "20px",
  fontXLarge: "33px",
  fontLarge: "26px",
  fontMedium: "20px",
  fontDefaultSize: "16px",
  fontSmall: "12.5px",
  fontDefaultFamily: "Georgia",
  fontDefaultWeight: "normal",
  colors: {
    primary: "lavender",
  },
  inputs: {
    borderRadius: "8px",
    padding: "8px",
  },
};

describe("Filters", () => {
  it("does not render if there are no filters", () => {
    const tree = create(<Filters />).toJSON();

    expect(tree).toBeNull();
  });

  it("does not render if filters are undefined", () => {
    const tree = create(<Filters type={undefined} />).toJSON();

    expect(tree).toBeNull();
  });

  it("renders given filters", () => {
    const tree = create(
      <ThemeProvider theme={theme}>
        <Filters types={filters} />
      </ThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("methods", () => {
    const testFilters = {
      test: ["info", "more info"],
      double: ["category", "another category"],
      better: ["range", "range plus", "range minus"],
    };
    const filters = { names: testFilters.test, activeFilters: [null] };

    it("creates a filter", () => {
      const filter = "test";

      const result = createFilter(filters, filter);

      expect(result).toHaveProperty("key", "Test");
      expect(result.props.onPress).toBeDefined();
    });

    it("creates filters for multiple names", () => {
      const result = createFilters(filters);

      expect(result).toHaveLength(2);
      expect(result[0]).toHaveProperty("key", "Info");
      expect(result[0].props.onPress).toBeDefined();
      expect(result[1]).toHaveProperty("key", "More info");
      expect(result[1].props.onPress).toBeDefined();
    });
  });
});
