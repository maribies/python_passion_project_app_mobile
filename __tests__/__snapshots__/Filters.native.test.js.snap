// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters renders given filters 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "flexWrap": "wrap",
        "marginBottom": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Quattrocento_400Regular",
          "fontSize": 20,
          "fontWeight": "normal",
          "marginBottom": 0,
          "marginRight": 20,
          "marginTop": 8,
        }
      }
    >
      Filters
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "lavender",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 8,
          "marginRight": 8,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "FanwoodText_400Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginBottom": 0,
          }
        }
      >
        Designers
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "lavender",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 8,
          "marginRight": 8,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "FanwoodText_400Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginBottom": 0,
          }
        }
      >
        Colors
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "lavender",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginBottom": 8,
          "marginRight": 8,
          "opacity": 1,
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "FanwoodText_400Regular",
            "fontSize": 16,
            "fontWeight": "normal",
            "marginBottom": 0,
          }
        }
      >
        Price
      </Text>
    </View>
  </View>
</View>
`;
